File: algorithms/D/Distribute Candies to People/Distribute Candies to People.java
class Solution {
    public int[] distributeCandies(int candies, int num_people) {
        int n=num_people;
        int a[]=new int[n];
        int k=1;
        while(candies>0){
            for(int i=0;i<n;i++){
                if(candies>=k){
                    a[i]+=k;
                    candies-=k;
                    k++;
                }
                else{
                    a[i]+=candies;
                    candies=0;
                    break;
                }
            }
        }
        return a;
    }
}

File: algorithms/D/Distribute Candies to People/Distribute Candies to People.cpp
class Solution {
public:
    vector<int> distributeCandies(int candies, int num_people) {
        vector<int> Candies (num_people, 0);
        int X = 0;
        while (candies)
        {
            for (int i = 0; i < num_people; ++i)
            {
                int Num = X * num_people + i + 1;
                if (candies >= Num)
                {
                    Candies[i] += Num;
                    candies -= Num;
                }
                else
                {
                    Candies[i] += candies;
                    candies = 0;
                    break;
                }
            }
            ++X;
        }
        return Candies;
    }
};
